department-of-veterans-affairs/caseflow
# Caseflow 3730 fixes

Resolves #[CASEFLOW-3730](https://vajira.max.gov/browse/CASEFLOW-3730)
This is fix of [CASEFLOW-3730's original pull request.](https://github.com/department-of-veterans-affairs/caseflow/pull/17202) The issue was that ChangeHearingRequestTypeTasks created in initial_tasks_factory.rb did not properly account for VSO permissions and types.
The original description for the story is below:
**Description**
As a VSO user I need the ability to access the convert-to-virtual form so that I am able to opt the Veteran/Appellant I represent into a Virtual hearing.

**Acceptance Criteria**

- [ ] VSO user has the following task action available for cases they represent: Convert hearing to virtual
- [ ] When the "Convert hearing to virtual" task action is selected the user is directed to the convert-to-virtual form 

**Testing Plan**
1. Open Caseflow app in localhost:3000
2. Change user to BVADWISE
3. Go to "Mail Intake" tab and start an intake
    a. Select VA form 10182 and continue to the next step.
    b. Enter 500000000 for the Veteran ID and continue to the next step.
    c. Select a date in the past for Receipt Date, select "Hearing" for the review option, and select Yes for "Is the claimant someone other than the Veteran?", then select Bob Vance. Bob Vance is a claimant with a VSO representative. The other option selections don't matter. Continue to the next step.
    d. Click on the "Add Issue" button and choose "Compensation" for Benefit Type and "Unknown Issue Type" for Issue Category. Choose a date in the past for Decision Date and fill in anything for the Issue description. Click on the "Add Issue" button, then click on the "Establish Appeal" button. 
4. Once the page says "Intake completed", click on "correct the issues"
5. Open a new tab and switch user to BVASYELLOW
6. The URL path should be in this form on the intake page: `/appeals/<appeal uuid>/edit`. 
    Copy and paste it into BVASYELLOW tab and change it to `/explain/appeals/<appeal uuid>`.
7. Click on the Task Tree option. The task tree of the new appeal should look like the image below:
![task tree](https://user-images.githubusercontent.com/93942793/169592867-659db03c-3e7a-4c80-be1a-d94fb92d606b.PNG)
